### PR TITLE
fix: Fix zstd decompression of multi-frame responses

### LINKED ISSUE #12234: ClientPayloadError: 400, message: Can not decode content-encoding: zstd - Already at the end of a Zstandard frame
### Describe the bug

When a server responds with zstd encoded response in multiple frames, current implementation fails decompressing the second frame [here](https://github.com/aio-libs/aiohttp/blob/7043bc563523992cee20754a59b1fd61f1972e7d/aiohttp/compression_utils.py#L345). Internally the exception is _Already at the end of a Zstandard frame_ but it ends up raising a _ClientPayloadError_.

Python docs mention this issue [here](https://docs.python.org/3/library/compression.zstd.html#compression.zstd.ZstdDecompressor:~:text=This%20class%20does%20not%20transparently%20handle%20inputs%20containing%20multiple%20compressed%20frames) and suggest to use [compression.zstd.decompress](https://docs.python.org/3/library/compression.zstd.html#compression.zstd.decompress) to avoid this limitation. I've tested internally with this change and works fine, but I'm not sure if this should be the way to go or you prefer another solution.

I have seen a similar issue into an unrelated project like [kafka-python](https://github.com/dpkp/kafka-python/issues/2716) just in case helps understanding it better.

PS: Unfortunately, I won’t be able to open an MR in the next two weeks, so feel free to fix it yourself if you’d like.

### To Reproduce

```python
import asyncio

import aiohttp
from compression.zstd import ZstdCompressor

BODY = b"A" * 50_000 + b"B" * 50_000


def zstd_frame(data: bytes) -> bytes:
    c = ZstdCompressor()
    return c.compress(data) + c.flush()


async def main() -> None:
    frame1 = zstd_frame(BODY[:50_000])
    frame2 = zstd_frame(BODY[50_000:])

    async def handle(reader, writer):
        while (await reader.readline()) != b"\r\n":
            pass
        writer.write(
            b"HTTP/1.1 200 OK\r\n"
            b"Content-Type: text/plain\r\n"
            b"Content-Encoding: zstd\r\n"
            b"Transfer-Encoding: chunked\r\n\r\n"
        )
        for frame in (frame1, frame2):
            writer.write(f"{len(frame):x}\r\n".encode() + frame + b"\r\n")
            await writer.drain()
            await asyncio.sleep(0.05)
        writer.write(b"0\r\n\r\n")
        await writer.drain()
        writer.close()

    server = await asyncio.start_server(handle, "127.0.0.1", 0)
    port = server.sockets[0].getsockname()[1]

    try:
        async with aiohttp.ClientSession() as session:
            async with session.get(f"http://127.0.0.1:{port}/") as resp:
                body = await resp.read()
                assert body == BODY
                print("OK")
    except aiohttp.ClientPayloadError as e:
        print(f"FAILED: {type(e).__name__}: {e}")
    finally:
        server.close()
        await server.wait_closed()


if __name__ == "__main__":
    asyncio.run(main())
```

### Expected behavior

No failures on multiframed zstd responses

### Logs/tracebacks

```python-traceback
FAILED: ClientPayloadError: 400, message:
  Can not decode content-encoding: zstd
```

### Python Version

```console
3.14
```

### aiohttp Version

```console
3.13.3
```

### multidict Version

```console
6.7.1
```

### propcache Version

```console
0.4.1
```

### yarl Version

```console
1.23.0
```

### OS

macOS

### Related component

Client

### Additional context

_No response_

### Code of Conduct

- [x] I agree to follow the aio-libs Code of Conduct

### PR DESCRIPTION
**Pull request by josumoreno-BP**
ZstdDecompressor is one-shot-per-frame: once a frame ends, subsequent decompress() calls raise EOFError. This broke HTTP responses where the server sends multiple zstd frames (common with chunked transfer encoding).

Detect frame boundaries via eof/unused_data attributes and create fresh decompressor instances for subsequent frames.

## What do these changes do?

Detect frame boundaries via eof/unused_data attributes and create fresh decompressor instances for subsequent frames.

## Are there changes in behavior for the user?

The change is seamless for the uses

## Is it a substantial burden for the maintainers to support this?

I think this is needed to cover multi-framed zstd responses. I tried to keep it simple but covering the cases that came to my mind.

## Related issue number

Fixes #12234 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes - I don't think we need to update the docs in this case
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES/` folder
  * name it `<issue_or_pr_num>.<type>.rst` (e.g. `588.bugfix.rst`)
  * if you don't have an issue number, change it to the pull request
    number after creating the PR
    * `.bugfix`: A bug fix for something the maintainers deemed an
      improper undesired behavior that got corrected to match
      pre-agreed expectations.
    * `.feature`: A new behavior, public APIs. That sort of stuff.
    * `.deprecation`: A declaration of future API removals and breaking
      changes in behavior.
    * `.breaking`: When something public is removed in a breaking way.
      Could be deprecated in an earlier release.
    * `.doc`: Notable updates to the documentation structure or build
      process.
    * `.packaging`: Notes for downstreams about unobvious side effects
      and tooling. Changes in the test invocation considerations and
      runtime assumptions.
    * `.contrib`: Stuff that affects the contributor experience. e.g.
      Running tests, building the docs, setting up the development
      environment.
    * `.misc`: Changes that are hard to assign to any of the above
      categories.
  * Make sure to use full sentences with correct case and punctuation,
    for example:
    ```rst
    Fixed issue with non-ascii contents in doctest text files
    -- by :user:`contributor-gh-handle`.
    ```

    Use the past tense or the present tense a non-imperative mood,
    referring to what's changed compared to the last released version
    of this project.

### CODE REVIEW

File: aiohttp/compression_utils.py [hunk 1]
```diff
@@ -342,7 +342,28 @@ def decompress_sync(
             if max_length == ZLIB_MAX_LENGTH_UNLIMITED
             else max_length
         )
-        return self._obj.decompress(data, zstd_max_length)
+        result = self._obj.decompress(data, zstd_max_length)
+
+        # Handle multi-frame zstd streams (RFC 8878 §3.1.1):
```
Comment (Dreamsorcerer) [inline aiohttp/compression_utils.py]:
```suggestion
        # Handle multi-frame zstd streams.
        # https://datatracker.ietf.org/doc/html/rfc8878#section-3.1.1
```

File: aiohttp/compression_utils.py [hunk 2]
```diff
@@ -342,7 +342,29 @@ def decompress_sync(
...
+        # ZstdDecompressor handles one frame only. When a frame ends,
+        # eof becomes True and any trailing data goes to unused_data.
+        # We create a fresh decompressor to continue with the next frame.
+        while self._obj.eof and self._obj.unused_data:
+            unused = self._obj.unused_data
+            self._obj = ZstdDecompressor()
+            if zstd_max_length != ZSTD_MAX_LENGTH_UNLIMITED:
+                zstd_max_length -= len(result)
+                if zstd_max_length <= 0:
+                    break
```
Comment (Dreamsorcerer) [inline aiohttp/compression_utils.py:363]: I think we're missing an edge case, which probably needs a new test.

If we have unused data here and break, we've ended up throwing it away. Presumably when it continues with further data, it'll fail as the input is corrupted.

**Comment by Dreamsorcerer** [comment]
Other than that subtle edge case, this looks very solid!

File: aiohttp/compression_utils.py (hunk 2 above)
Comment (josumoreno-BP) [reply aiohttp/compression_utils.py:363]: Added a commit on top, let me know wdyt

Comment (Dreamsorcerer) [reply aiohttp/compression_utils.py:363]: Seems reasonable.

To sanity check my own changes:
https://github.com/aio-libs/aiohttp/pull/11966/changes#diff-c88fdeb279a2914aa157a5ebbf863d092eeb3014be8e8e4a1eaef48daa186651R377

That line will need to change to:
```
    def data_available(self) -> bool:
        return (not self._obj.needs_input and not self._obj.eof) or self._pending_unused_data
```
if I'm understanding correctly.

Comment (josumoreno-BP) [reply aiohttp/compression_utils.py:363]: Yes!
